pkg/controller/route/route_controller.go

```diff
@@ -69,6 +69,37 @@ type RouteController struct {
 	recorder         record.EventRecorder
 }
 
+// routesState contains a map of podCIDRs and their states belonging to each node
+type routesState struct {
+	sync.RWMutex
+
+	routes map[types.NodeName]map[string]bool
```

Comment:
should we abstract `map[string]bool`with a struct that has a podCIDR field and if it was created or not ?

 this would give something like:

```
routes map[types.NodeName][]PodCidrInfo
```

Reply:
Maybe, this PR is mainly clean up and following up to some prior comments in https://github.com/kubernetes/kubernetes/pull/73977, would want to keep that change in a separate PR.

Reply:
ack. can we at least add a todo for this ?